### PR TITLE
Cart Line Items » cartItemClass: Add code snippet and screenshot

### LINKED ISSUE #10899: `Cart Line Items » cartItemClass`: Add code snippet and screenshot
Recently, we've updated the "Available Extensibility Interfaces for the Cart and Checkout blocks" post. In this post, we introduced new code snippets and screenshots. This issue aims to add the following code snippet and screenshot to the [`Cart Line Items » cartItemClass`](https://github.com/woocommerce/woocommerce-blocks/blob/6083484477410fd99d1efd5ca909288eb4d19963/docs/third-party-developers/extensibility/checkout-block/available-filters.md?plain=1#L26) section:

```js
const { registerCheckoutFilters } = window.wc.blocksCheckout;
 
// Adjust cart item class of the cart line items.
registerCheckoutFilters( 'example-extension', {
  cartItemClass: ( value, extensions, args ) => {
    // Return early since this filter is not being applied in the Cart context.
    // We must return the original value we received here.
    if ( args?.context !== 'cart' ) {
      return value;
    }
    return 'my-custom-class';
  }
} );
```

screenshot-2023-08-24-at-13 07 26

---

> **Note**
> The code snippet and screenshot should be placed below the table with the filter names and the sentence _"Each of these filters has the following arguments [...]"_, and should have the following subheading
```md
### cartItemClass
```

### PR DESCRIPTION
**Pull request by WebVishnu**
## What
Code snippet and a screenshot is required at [Cart Line Items » cartItemClass](https://github.com/woocommerce/woocommerce-blocks/blob/6083484477410fd99d1efd5ca909288eb4d19963/docs/third-party-developers/extensibility/checkout-block/available-filters.md?plain=1#L26)
Fixes #10899

## Why
Due to the updated "Available Extensibility Interfaces for the Cart and Checkout blocks" post

## Testing Instructions
A reviewer should look up the changed *.md file and ensure that the added code and the screenshot match with the ones mentioned in #10899

_Please consider any edge cases this change may have, and also other areas of the product this may impact._
N/A

* [ ] Do not include in the Testing Notes 
* [ ] Should be tested by the development team exclusively 

## Screenshots or screencast
N/A

| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility
WooCommerce Core: The change will be visible to everyone who is using the [WooCommerce](https://wordpress.org/plugins/woocommerce/) plugin.
Feature plugin: The change will only be visible for people that are using the [WooCommerce Blocks](https://wordpress.org/plugins/woo-gutenberg-products-block/) featured plugin.
Experimental: The change will only in development mode.
N/A: The change will not be visible, such as updating the documentation.

Required:

* [ ] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [x] N/A

## Checklist

Required:
* [X] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [ ] This PR is assigned to a milestone.

Conditional:
* [ ] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

> Add suggested changelog entry here.

### CODE REVIEW

File: docs/third-party-developers/extensibility/checkout-block/available-filters.md [hunk 1]
```diff
@@ -29,6 +29,24 @@ Line items refer to each item listed in the cart or checkout. For instance, the
 
 ![Cart Line Items](https://user-images.githubusercontent.com/17236129/192795735-bd6e8494-e5c9-4384-b6a4-092a579c4755.png)
 
+```
```
Comment (nielslange) [inline docs/third-party-developers/extensibility/checkout-block/available-filters.md]:
```suggestion
```js
```

We usually explicitly specify the language used within fenced code blocks. That way, we can ensure that GitHub is applying the correct syntax highlighting.

**Review by nielslange — COMMENTED** [review]
Hello @WebVishnu 👋 Thank you very much for your first contribution to the WooCommerce Blocks repo! 🎉

I briefly looked at the PR and noticed a few things, for which you can find my initial feedback below.

---

In the description of this PR, you mentioned the following:
> https://github.com/woocommerce/woocommerce-blocks/issues/10899 - Code snippet and a screenshot is required at [Cart Line Items » cartItemClass](https://github.com/woocommerce/woocommerce-blocks/blob/6083484477410fd99d1efd5ca909288eb4d19963/docs/third-party-developers/extensibility/checkout-block/available-filters.md?plain=1#L26)
Fixes #

Fixes is one of the keywords that connects a PR to an issue. On https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue, you can find detailed information on this. I see that you mentioned that corresponding issue number. However, without the keyword, GitHub can not link your PR to the issue. I suggest changing your description as follows:
> Code snippet and a screenshot is required at [Cart Line Items » cartItemClass](https://github.com/woocommerce/woocommerce-blocks/blob/6083484477410fd99d1efd5ca909288eb4d19963/docs/third-party-developers/extensibility/checkout-block/available-filters.md?plain=1#L26)
Fixes #10899

---

I see that no testing instructions have been mentioned. You could mention that a reviewer should look up the changed *.md file and ensure that the added code and the screenshot match with the ones as mentioned in https://github.com/woocommerce/woocommerce-blocks/issues/10899.

---

I see that the `Screenshots or screencast` section is also empty. You can remove sections that are not relevant to a PR. 

---

I've checked the checkbox `N/A` in the section `WooCommerce Visibility`. This section explains where a certain change will be visible. The following list explains the various options:

- WooCommerce Core: The change will be visible for everyone who is using the [WooCommerce](https://wordpress.org/plugins/woocommerce/) plugin. 
- Feature plugin: The change will only be visible for people that are using the [WooCommerce Blocks](https://wordpress.org/plugins/woo-gutenberg-products-block/) featured plugin. 
-  Experimental: The change will only in development mode.
-  N/A: The change will not be visible, such as updating the documentation.

---

I see that the checkbox `This PR is assigned to a milestone.` in the section `Checklist` is checked, but the PR has no assigned milestone. When in doubt, it's better not to check any checkboxes, as some of them trigger actions in the background. Checking a wrong checkbox, might lead to an unexpected side effect. 

---

I see that the section `Changelog` is empty. You can remove this section completely. I'll add the label `skip-changelog` to highlight that this PR will not contain a changelog entry.

---

Finally, when working on a ticket, could you please leave a brief note in the corresponding issue, saying that you like to work on it? We can then assign the ticket to you and avoid that multiple people accidentally working on the same issue. Could you also leave a brief note on https://github.com/woocommerce/woocommerce-blocks/issues/10899 so that I can assign that issue officially to you?

---

The title of this PR currently says
> Fix - Code snippet and screenshot added

Given that we'll update more sections, I would appreciate it if you could provide a more speaking title. For example, you could simply use the same title as the corresponding issue:
> `Cart Line Items » cartItemClass`: Add code snippet and screenshot

---

That's all from my side so far. And, again, thank you so much for contributing to WooCommerce Blocks! Your contribution is very much appreciated! 🙌

File: docs/third-party-developers/extensibility/checkout-block/available-filters.md (hunk 1 above)
Comment (WebVishnu) [reply docs/third-party-developers/extensibility/checkout-block/available-filters.md]: Thanks for all the suggestions @nielslange for my PR. I will definitely make a few changes according to your needs. 
Please assign this issue to me.

Comment (nielslange) [reply docs/third-party-developers/extensibility/checkout-block/available-filters.md]: > Thanks for all the suggestions @nielslange for my PR. I will definitely make a few changes according to your needs.

Great to hear. And if you need any assistance, please don't hesitate to reach out. 🙌

> Please assign this issue to me.

I just assigned this issue to you, @WebVishnu.

Comment (WebVishnu) [reply docs/third-party-developers/extensibility/checkout-block/available-filters.md]: Sorry to bother you @nielslange  but I have an issue. 
I have edited the comment, What should be my next step?

Comment (nielslange) [reply docs/third-party-developers/extensibility/checkout-block/available-filters.md]: > Sorry to bother you @nielslange but I have an issue. I have edited the comment, What should be my next step?

Don't worry, @WebVishnu. I'm happy to assist. You mentioned that you edited the comment and I see most of the changes. The most important one is that you mentioned `Fixes #10899`, which connected your PR to the original issue. Thanks for doing that.

The next step would be to specify the language of the code snippet. Above, I suggested the change. There, you should see a button that says `Commit suggestion`:

Screenshot 2023-09-13 at 13 37 02

When you click the `Commit suggestion` button, my suggestion will automatically be added to your PR. On https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request, you can find further information on this feature.

I hope this helps, and please let me know if you need any further assistance.

**Comment by WebVishnu** [comment]
Hey, @nielslange, I have merged the suggestions. 
Let me know if there if you need any change from my side

**Comment by nielslange** [comment]
> Hey, @nielslange, I have merged the suggestions. Let me know if there if you need any change from my side

Looking at the final result, I noticed that the code example and the screenshot had been placed above the table with the filter names. This was my fault. I should have specified that. 

We should move both the code example and the screenshot below the table respectively below the sentence _"Each of these filters has the following arguments passed to it: { context: 'cart', cartItem: CartItem } ([CartItem](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/c00da597efe4c16fcf5481c213d8052ec5df3766/assets/js/type-defs/cart.ts#L113))"_ and add the following title:
```
### cartItemClass
```

This will create a third-level heading, similar to an `<h3>` HTML element. Once you applied this change, I'd say your PR can be approved and merged. 😀

**Review by nielslange — COMMENTED** [review]
Hello @WebVishnu, I just wanted to ask if my comment in https://github.com/woocommerce/woocommerce-blocks/pull/10903#issuecomment-1717281894 was clear or if you need any assistance to proceed with your PR. 😀

**Comment by ralucaStan** [comment]
Looks like @WebVishnu is not answering anymore. @nielslange would you mind taking care of getting this PR merged?

**Comment by nielslange** [comment]
> Looks like @WebVishnu is not answering anymore. @nielslange would you mind taking care of getting this PR merged?

Thanks for reaching out, @ralucaStan. I missed that this PR was still open, and @masteradhoc already took care of it in #10989.

**Pull request closed without merging**